src/components/CryptoFinder.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { useState, useEffect } from "react";
+import { useQuery } from "@tanstack/react-query";
+import axios from "axios";
+import CryptoCard from "./CryptoCard";
+import CoinGeckoResponse from "../models/CoinGeckoResponse";
+import "./CryptoFinder.css";
+
+type CryptoList = {
+  id: string;
+  symbol: string;
+  name: string;
+};
+
+type CryptoInfo = {
+  id: string;
+  name: string;
+};
+
+const fetchCryptoData = async (id: string) => {
+  const response = await axios.get<CoinGeckoResponse>(
+    `https://api.coingecko.com/api/v3/simple/price?ids=${id}&vs_currencies=usd,eur,gbp&precision=2`
+  );
+  return response.data[id];
+};
+
+const fetchCryptoList = async (): Promise<CryptoList[]> => {
+  const response = await axios.get<CryptoList[]>(
+    `https://api.coingecko.com/api/v3/coins/list`
+  );
+  return response.data;
+};
```

Comment:
you could move these into apiClients/crypto.api.ts

Reply:
Done, in the latest commit